ivy/functional/ivy/experimental/layers.py

```diff
@@ -180,9 +205,45 @@ def max_unpool1d(
            [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0., 16.,  0., 17.,  0.],
             [ 0., 18.,  0., 19.,  0.,  0.,  0.,  0., 20.,  0., 21.,  0.]]])
     """
-    return ivy.current_backend(x).max_unpool1d(
-        x, indices, kernel, strides, padding, data_format=data_format, out=out
+    if strides is None:
+        strides = kernel_size
+    input_shape = input.shape
+    if data_format in ["NCW", "NWC"]:
+        revert = False
+        if data_format == "NWC":
+            x_len = (input_shape[1],)
+            input = input.permute_dims((0, 2, 1))
+            indices = indices.permute_dims((0, 2, 1))
+            revert = True
+        else:
+            x_len = (input_shape[-1],)
+    else:
+        raise ValueError(
+            f"data_format attr should be NCW or NWC but found {data_format}"
+        )
+    input_shape = input.shape
+    kernel_size = _broadcast_pooling_helper(kernel_size, "1d", name="kernel_size")
+    padding = _broadcast_pooling_helper(padding, "1d", name="padding")
+    strides = _broadcast_pooling_helper(strides, "1d", name="strides")
+    output_len = _cal_output_shape(x_len, padding, kernel_size, strides)
+    output_shape = list(input_shape[:-1]) + output_len
+    one_like_mask = ivy.ones_like(indices, dtype="int32")
+    batch_shape = [input_shape[0], 1, 1]
+    batch_range = ivy.reshape(
+        ivy.arange(0, output_shape[0], dtype="int32"), batch_shape
     )
+    b = one_like_mask * batch_range
+    feature_range = ivy.arange(0, output_shape[1], dtype="int32").reshape(1, -1, 1)
+    f = one_like_mask * feature_range
+    updates_size = input.size
+    indices = ivy.stack([b, f, indices]).reshape(3, updates_size)
+    output = ivy.zeros(output_shape, dtype=input.dtype)
+    indices = tuple(indices)
+    values = input.reshape((-1,))
+    output[indices] = values
+    if revert:
+        output = output.permute_dims([0, 2, 1])
+    return output
 
```

Comment:
The code looks good overall. The only thing remaining is to add the partial torch backend.

Reply:
What do you mean by partial torch backend? I did not get what you meant here

Reply:
https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#partial-mixed-functions